pallets/exchange/benchmarking/src/lib.rs

```diff
@@ -119,6 +119,8 @@ benchmarks! {
 	verify {
 	}
 
+	// REVIEW: This is not testing the worst case. The worst case is
+	// when appending to a long list of intentions.
```

Comment:
Working in improving the benchamrks now  but I hit an issue which I would appreciate your help.

This sell benchmark  - the worst case scenario is to append an intention to list where is already many intentions.

this is fine. However, I noticed that feeding those intentions is causing a performance issue. I investigated and found out that it is caused by the resizing of this vector. 

And it seems that the capacity is extended by n items - n is rather small ( it looks like 4 but i not sure 100% here ). 
So every n-th intentions the vector is resized. So creating 10_000 intentions is taking therefore lot of time.

I tried to create the initial list  with larger capacity, so the resizing would happen less often. 

And here is my issue - When i stored this vec in the storage, the capacity does not seem to be preserved. And appending next intentions i am back to square one.

Any hint/advice what I can do here? thanks in advance.

Reply:
I imagine (would need to check to make 100% sure) that the vec is serialized according to number of elements, not the capacity.
But if it's just about the benchmark you should be able to directly set the storage vec to whatever limit you want to test.
So create a vec with e.g. 1000 intentions and just put it into storage directly. Would that work?